libcudacxx/codegen/codegen.cpp

```diff
@@ -326,30 +333,52 @@ _LIBCUDACXX_BEGIN_NAMESPACE_STD
             }
             for (auto& cv : cv_qualifier)
             {
+              out << "template<class _Type, _CUDA_VSTD::__enable_if_t<sizeof(_Type)==" << sz / 8;
+              if (type.first == "f")
+              {
+                out << " && _CCCL_TRAIT(is_floating_point, _Type), int> = 0>\n";
+              }
+              else if (rmw.first == "fetch_max" || rmw.first == "fetch_min")
+              {
+                if (type.first == "u")
+                {
+                  out << " && _CCCL_TRAIT(is_integral, _Type) && _CCCL_TRAIT(is_unsigned, _Type), int> "
+                         "= 0>\n";
+                }
+                else if (type.first == "s")
+                {
+                  out << " && _CCCL_TRAIT(is_integral, _Type) && _CCCL_TRAIT(is_signed, _Type), int> = "
```

Comment:
These shouldn't need `is_integral`, no? Just `is_{,un}signed`?

Reply:
We need to differentiate from floating point types in this case. This code is unchanged. It was just moved from below.